tests/python/unittest/test_thread_local.py

```diff
@@ -133,6 +133,19 @@ def f():
     thread.join()
     event.clear()
     assert status[0], "Spawned thread isn't using the correct blockscope namemanager"
+    
+def test_createblock():
+    status = [False]
+    def f():
+        net = mx.gluon.nn.Dense(2)
+        net.initialize()
+        net(mx.nd.array([1, 2, 3]))
+        status[0] = True
+
+    thread = threading.Thread(target=f)
+    thread.start()
+    thread.join()
+    assert status[0], "Failed to create a layer within a thread"
```

Comment:
Can you also add the test for running the following functions inside a thread:

```
  def g():
      data = mx.sym.Variable('data', attr={'a': 'b'})
```

```
  def f():
      a = mx.sym.var("a")
      b = mx.sym.var("b")
      a_ = mx.nd.ones((2, 2))
      c_ = a_.copy()
      func1 = (a + b).bind(mx.cpu(), args={'a': a_, 'b': c_})
      func1.forward()[0].wait_to_read()
```

Reply:
Will do :+1: